source/common/http/http2/codec_impl.cc

```diff
@@ -2011,9 +2024,12 @@ Status ClientConnectionImpl::trackInboundFrames(const nghttp2_frame_hd* hd,
     ENVOY_CONN_LOG(trace, "error reading frame: {} received in this HTTP/2 session.", connection_,
                    result.message());
     if (isInboundFramesWithEmptyPayloadError(result)) {
-      ConnectionImpl::StreamImpl* stream = getStream(hd->stream_id);
+      ConnectionImpl::StreamImpl* stream = getStreamUnchecked(hd->stream_id);
       if (stream) {
```

Comment:
@yanavlasov would like to hear your thoughts as original code author.
Is the returned `stream` always expected to be non-null? If so, should this if statement change to an assert?

Context: the else-clause in this PR is failing coverage report, and @vehre-x41 could not find a way to trigger this path.

Reply:
For client codec the stream can never be nullptr here, since it is created right before the codec allocates stream_id. The codec would have failed before this point if it got response for unknown stream_id.

To be  on the defensive side I think the else statement can simply be omitted here. Maybe just add a comment that the `if` statement is defensive.